hooks/command

```diff
@@ -29,6 +29,11 @@ readonly log_complete_timeout_sec="${BUILDKITE_PLUGIN_K8S_LOG_COMPLETE_TIMEOUT_S
 readonly use_agent_node_affinity=${BUILDKITE_PLUGIN_K8S_USE_AGENT_NODE_AFFINITY:-false}
 readonly print_resulting_job_spec=${BUILDKITE_PLUGIN_K8S_PRINT_RESULTING_JOB_SPEC:-false}
 readonly jobs_cleanup_via_plugin=${BUILDKITE_PLUGIN_K8S_JOBS_CLEANUP_VIA_PLUGIN:-}
+readonly skip_on_macos=${BUILDKITE_PLUGIN_K8S_SKIP_ON_MACOS:-}
+
+if [[ "$skip_on_macos" == "true" && $OSTYPE == darwin* ]]; then
```

Comment:
Not sure I follow the issue here could you clarify why we want to do this? the plugin is supposed to only be run on kubernetes so atm I can't see why we'd want to support running it on bare macos to begin with

Reply:
The problem is that if you have some basic job config like this:

``yaml
steps:
  - command: "echo 'Hello, World!'"
    plugins:
      - EmbarkStudios/k8s:
          image: alpine
```

And you have both macOS and Linux agents, if the macOS agents accept the job, without a chance like this on it will try to do all of this logic, which likely doesn't make sense on macOS as you say. The issue is you can't do something like "plugins only for linux"

We have this setup where we have both macOS and Linux agents on the same queue because we support iOS builds, and it helps us avoid CI downtime because jobs can pick either agent (although we give priority to Linux agents)

Reply:
I see, I think thats reasonable then